azurerm/resource_arm_storage_blob.go

```diff
@@ -564,6 +579,15 @@ func resourceArmStorageBlobUpdate(d *schema.ResourceData, meta interface{}) erro
 		return fmt.Errorf("Error setting properties of blob %s (container %s, storage account %s): %+v", id.blobName, id.containerName, id.storageAccountName, err)
 	}
 
+	if d.HasChange("metadata") {
+		blob.Metadata = expandBlobMetadata(d)
```

Comment:
As these functions are package wide could we rename this to
```suggestion
		blob.Metadata = expandStorageAccountBlobMetadata(d)
```